x/ccv/provider/keeper/keeper.go

```diff
@@ -1248,6 +1248,55 @@ func (k Keeper) SetToBeOptedIn(
 	store.Set(types.ToBeOptedInKey(chainID, providerAddr), []byte{})
 }
 
+// SetConsumerCommissionRate sets a per-consumer chain commission rate
+// for the given validator address
+func (k Keeper) SetConsumerCommissionRate(
+	ctx sdk.Context,
+	chainID string,
+	providerAddr types.ProviderConsAddress,
+	commissionRate sdk.Dec,
+) {
+	store := ctx.KVStore(k.storeKey)
+	bz, err := commissionRate.Marshal()
+	if err != nil {
+		panic(fmt.Errorf("consumer commission rate marshalling failed:%s", err))
+	}
+
+	store.Set(types.ConsumerCommissionRateKey(chainID, providerAddr), bz)
+}
+
+// GetConsumerCommissionRate returns the per-consumer commission rate set
+// for the given validator address
+func (k Keeper) GetConsumerCommissionRate(
+	ctx sdk.Context,
+	chainID string,
+	providerAddr types.ProviderConsAddress,
+) (sdk.Dec, bool) {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.ConsumerCommissionRateKey(chainID, providerAddr))
+	if bz == nil {
+		return sdk.ZeroDec(), false
+	}
+
+	cr := sdk.Dec{}
+	if err := cr.Unmarshal(bz); err != nil {
+		panic(fmt.Errorf("consumer commission rate unmarshalling failed:%s", err))
```

Comment:
nit: ```suggestion
		panic(fmt.Errorf("consumer commission rate unmarshalling failed: %s", err))
```

Reply:
Avoid panicking and either return an error or false. If there is an issue with unmarshaling, log an error and use the default commission rate.